modules/service/src/main/scala/lucuma/odb/service/ObscalcService.scala

```diff
@@ -196,28 +206,43 @@ object ObscalcService:
         pending: Obscalc.PendingCalc
       )(using NoTransaction[F]): F[Obscalc.Result] =
 
+        def sequenceUnavailable(msg: String): OdbError =
+          OdbError.SequenceUnavailable(pending.observationId, s"Could not generate a sequence for ${pending.observationId}: $msg".some)
+
         val params: EitherT[F, OdbError, GeneratorParams] =
           EitherT:
             services.transactionally:
               generatorParamsService
                 .selectOne(pending.programId, pending.observationId)
-                .map(_.leftMap(e => OdbError.SequenceUnavailable(pending.observationId, s"Could not generate a sequence for ${pending.observationId}: ${e.format}".some)))
-
-        def digest(itcResult: Either[OdbError, ItcService.AsterismResults]): EitherT[F, OdbError, ExecutionDigest] =
-          for
+                .map(_.leftMap(e => sequenceUnavailable(e.format)))
+
+        def workflow(
+          itc: Option[ItcService.AsterismResults],
+          dig: Option[ExecutionDigest]
+        ): F[ObservationWorkflow] =
+          services
+            .transactionally:
+              observationWorkflowService.getCalculatedWorkflow(pending.observationId, itc, dig.map(_.science.executionState))
+            .flatMap: r =>
+              r.toOption.fold(Logger[F].warn(s"Failure calculating workflow: $r").as(UndefinedWorkflow))(_.pure[F])
```

Comment:
I'm unsure about this.  The workflow service returns an `ObservationWorkflow` in a `Result` wrapper.  I'm not sure when the `Result` might be anything other than success (particularly since no ITC call is being made in the `getCalculatedWorkflow` version).